repair/repair.cc

```diff
@@ -1492,8 +1492,34 @@ future<> repair::data_sync_repair_task_impl::run() {
     auto germs = make_lw_shared(co_await locator::make_global_effective_replication_map(sharded_db, keyspace));
 
     auto id = get_repair_uniq_id();
-    rlogger.info("repair[{}]: sync data for keyspace={}, status=started", id.uuid(), keyspace);
-    co_await module->run(id, [this, &rs, id, &db, keyspace, germs = std::move(germs), &ranges = _ranges, &neighbors = _neighbors, reason = _reason] () mutable {
+    bool small_table_optimization = false;
+
+    static const std::unordered_set<sstring> small_table_optimization_enabled_ks = {
+        "system_distributed",
+        "system_distributed_everywhere",
+        "system_auth",
+        "system_traces"
+    };
+    if (_reason == streaming::stream_reason::bootstrap ||
+        _reason == streaming::stream_reason::decommission) {
```

Comment:
Why only for bootstrap and decomission? Wouldn't it be beneficial for other reasons too?

Reply:
This is a starter for most common use cases. We can enable more as follow ups.